### PR TITLE
Error registry

### PR DESCRIPTION
**Pull request by huitema**
First cut at creating an error registry.

### CODE REVIEW

File: draft-ietf-dprive-dnsoquic.md [hunk 1]
```diff
@@ -893,6 +893,67 @@ the Extended DNS Error Codes registry {{!RFC8914}}:
...
+Provisional registrations of codepoints are intended to allow for private use
+and experimentation with extensions to DNS over QUIC. Provisional registrations
+only require the inclusion of the codepoint value and contact information. However,
+provisional registrations could be reclaimed and reassigned for another purpose.
+Provisional registrations require Expert Review, as defined in Section 4.5 of {{!RFC8126}}.
+The designated expert or experts are advised that only registrations for an excessive
+proportion of remaining codepoint space or the very first unassigned value (see Section 22.1.2)
+can be rejected. Provisional registrations will include a Date field that indicates
+when the registration was last updated. A request to update the date on any provisional
+registration can be made without review from the designated expert(s).
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:917]: You cite Section 22.1.2 here, but I think that is down to copy and paste from QUIC.  Maybe you can just reference QUIC and cut down this text.

File: draft-ietf-dprive-dnsoquic.md [hunk 2]
```diff
@@ -893,6 +893,67 @@ the Extended DNS Error Codes registry {{!RFC8914}}:
...
+
+Registrations in this registry MUST include the following fields:
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:919]: Is this just the permanent registrations?  Otherwise it will contradict the statement above.

**Review by martinthomson — COMMENTED** [review]
Seems like the right direction.

File: draft-ietf-dprive-dnsoquic.md (hunk 2 above)
Comment (huitema) [reply draft-ietf-dprive-dnsoquic.md:919]: @martinthomson Thanks for the suggestions. I have revised the text, hopefully for greater clarity.

**Review by martinthomson — APPROVED** [review]

**Comment by saradickinson** [comment]
Looks good. 2 minor suggestions:
* We could add a reference to this section from `Section 5.3.  DoQ Error Codes`
* We could add text at the start of this section something like 'In accordance with recommended practice [ref]....`  so we don't get lots of questions about why we are doing it this way?

**Review by saradickinson — APPROVED** [review]

**Comment by huitema** [comment]
@saradickinson adding a reference to the error registry in Section 5.3. is easy, I did that. But I don't have a reference at hand on best practices for registries. I suppose I could cite RFC 9000. Is that what you meant?

**Comment by huitema** [comment]
OK, In the interest of not wasting time, I just added a comment in the registry section, "(This design is aligned with the principles
set in section 22 of {{!RFC9000}}.)"

**Pull request merged**